Content.Shared/Movement/Systems/SpeedModifierContactsSystem.cs

```diff
@@ -95,6 +97,11 @@ private void MovementSpeedCheck(EntityUid uid, SpeedModifiedByContactComponent c
                 if (_whitelistSystem.IsWhitelistPass(slowContactsComponent.IgnoreWhitelist, uid))
                     continue;
 
+                // unless the entity applying the contact speed modifier is explicitly set to affect airborne entities
+                // entities that are weightless or in the air, i.e. no gravity or flying mobs, should not be affected by contact speed modifiers
+                if (!slowContactsComponent.AffectAirborne && (_gravity.IsWeightless(uid) || physicsComponent.BodyStatus == BodyStatus.InAir))
+                    continue;
```

Comment:
You should cache this isweightless check somewhere so it isn't re-checked for every contact as it's expensive.

Reply:
Done, let me know if it's alright now or still needs further changes.